aws/ci_build

```diff
@@ -101,6 +126,10 @@ def build
 
     status
   end
+ensure
```

Comment:
This ensure block is several lines way from the logic we want to wrap in a lock/release and duplicates some existing code above. See my thoughts above about consolidating this.